code/Pawn/Worm.cs

```diff
@@ -1,13 +1,17 @@
 ﻿using Sandbox;
 using System.Linq;
 using TerryForm.Weapons;
+using TerryForm.States;
 
 namespace TerryForm.Pawn
 {
 	public partial class Worm : Sandbox.Player
 	{
 		[Net] public Weapon EquippedWeapon { get; set; }
 
+		// We should probably compare PlayerID's or something like that rather than comparing the entities directly.
+		public bool IsMyTurn => (Game.StateHandler?.State as PlayingState)?.Turn?.ActivePlayer?.ActiveWorm?.Equals( this ) ?? false;
```

Comment:
Not a fan of this, is there no better way to handle it?

Reply:
I'll come back to this one when we figure out the turn stuff.

Reply:
Should be handled now.